### DISCUSSION TITLE
OpenID with other account causes duplicate key error

### DISCUSSION THREAD

**Discussion by mschmidwalnuts** [danny-avila/LibreChat#4590] [Troubleshooting]
I have a LibreChat instance hosted on Azure. I'm using our OpenID (Azure Entra) to provide authentication. After setting up a clean instance, I logged in with one account (A). That worked without an issue. Now everytime I try to login with a different account (B) the following error is generated:

```
MongoServerError: E11000 duplicate key error collection: test.users. Failed _id or unique index constraint.
    at /app/node_modules/mongodb/lib/operations/insert.js:50:33
    at /app/node_modules/mongodb/lib/operations/command.js:84:64
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

Configuration:
OPENID_CALLBACK_URL=/oauth/openid/callback
OPENID_CLIENT_ID=redacted
OPENID_CLIENT_SECRET=redacted
OPENID_ISSUER=https://login.microsoftonline.com/redacted/v2.0/
OPENID_SCOPE=openid profile email
OPENID_SESSION_SECRET=redacted

ALLOW_REGISTRATION=false
ALLOW_SOCIAL_REGISTRATION=false
ALLOW_EMAIL_LOGIN=false
ALLOW_SOCIAL_LOGIN=true
ALLOW_UNVERIFIED_EMAIL_LOGIN=true

I'm running on the latest image of librechat (ghcr.io/danny-avila/librechat-dev-api:latest)

**Comment by danny-avila** [comment]
Are you using Azure Cosmos DB?

**Reply by mschmidwalnuts** [reply]
Yes

**Reply by danny-avila** [reply]
That's the issue. It's not fully compatible unfortunately.

**Reply by danny-avila** [reply]
There might be a MongoDB driver option that I'm not too familiar with, that I've heard people have had success with.

**Reply by mschmidwalnuts** [reply]
Thanks! I will try the MongoDB Atlas Ressource from Azure Marketplace to see if the error persists
